app/src/main/java/org/mozilla/fenix/settings/DataChoicesFragment.kt

```diff
@@ -20,23 +19,18 @@ import org.mozilla.fenix.ext.showToolbar
  */
 class DataChoicesFragment : PreferenceFragmentCompat() {
 
-    private val preferenceChangeListener =
-        SharedPreferences.OnSharedPreferenceChangeListener { sharedPreferences, key ->
-            when (key) {
-                getPreferenceKey(R.string.pref_key_telemetry) -> {
-                    if (sharedPreferences.getBoolean(key, requireContext().settings().isTelemetryEnabled)) {
-                        context?.components?.analytics?.metrics?.start()
-                    } else {
-                        context?.components?.analytics?.metrics?.stop()
-                    }
-                }
-            }
-        }
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        context?.let {
-            preferenceManager.sharedPreferences.registerOnSharedPreferenceChangeListener(preferenceChangeListener)
+
+        val context = requireContext()
+        preferenceManager.sharedPreferences.registerOnSharedPreferenceChangeListener(this) { sharedPreferences, key ->
+            if (key == getPreferenceKey(R.string.pref_key_telemetry)) {
```

Comment:
I have a _slight_ preference for doing `if (key != getPreferenceKey(R.string.pref_key_telemetry)) { return }` just to avoid nesting these if statements here. I don't feel strongly on it though if you prefer to keep it this way :)

Reply:
After making the change the compiler started complaining about return in the lambda. I'm going to leave the if statement since I think its more readable than `return@foo`